trustymail/domain.py

```diff
@@ -85,28 +85,33 @@ def add_mx_record(self, record):
         self.mail_servers.append(record.exchange.to_text().rstrip('.').lower())
 
     def parent_has_dmarc(self):
-        if self.base_domain is None:
-            return None
-        return self.base_domain.has_dmarc()
+        ans = self.has_dmarc()
+        if self.base_domain:
+            ans = self.base_domain.has_dmarc()
+        return ans
 
     def parent_valid_dmarc(self):
-        if self.base_domain is None:
-            return None
-        return self.base_domain.valid_dmarc
+        ans = self.valid_dmarc
+        if self.base_domain:
+            return self.base_domain.valid_dmarc
+        return ans
 
     def parent_dmarc_results(self):
-        if self.base_domain is None:
-            return None
-        return self.format_list(self.base_domain.dmarc)
+        ans = self.format_list(self.dmarc)
+        if self.base_domain:
+            ans = self.format_list(self.base_domain.dmarc)
+        return ans
 
     def get_dmarc_policy(self):
-        # If the policy was never set, or isn't in the list of valid policies, check the parents.
-        if self.dmarc_policy is None or self.dmarc_policy.lower() not in ['quarantine', 'reject', 'none']:
-            if self.base_domain is None:
-                return ''
+        ans = self.dmarc_policy
+        # If the policy was never set, or isn't in the list of valid
+        # policies, check the parents.
+        if ans is None or ans.lower() not in ['quarantine', 'reject', 'none']:
+            if self.base_domain:
+                ans = self.base_domain.get_dmarc_policy()
             else:
-                return self.base_domain.get_dmarc_policy()
-        return self.dmarc_policy
+                ans = ''
```

Comment:
Do we want an empty string or `None` here?

Reply:
@konklone, I can't add you as a reviewer, but please feel free.  You created the issue, after all. :+1:

Reply:
If this intended to indicate "nothing", then I would say `None`. This gives clients a more precise answer, and the ability to distinguish between a situation where it's missing and a situation where the value is `""`.